cpp/src/strings/json/json_path.cu

```diff
@@ -0,0 +1,460 @@
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/get_value.cuh>
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <rmm/exec_policy.hpp>
+
+#include <cudf_test/column_wrapper.hpp>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+
+namespace {
+
+using namespace cudf;
+
+CUDA_HOST_DEVICE_CALLABLE bool device_strncmp(const char* str1, const char* str2, size_t num_chars)
```

Comment:
Does a function that is prefixed `device_` ever need to be called from the host? If not, then don't make it host device callable, make it device callable. If it does, then rename it.

Reply:
I have this setup currently so that I can invoke it from the CPU for debugging purposes. Considerably easier to debug parsing code like this with actual breakpoints :)